cpp/src/parquet/stream_writer.h

```diff
@@ -153,20 +189,29 @@ class PARQUET_EXPORT StreamWriter {
   void CheckColumn(Type::type physical_type, ConvertedType::type converted_type,
                    int length = -1);
 
+  void SkipOptionalColumn();
+
  private:
   using node_ptr_type = std::shared_ptr<schema::PrimitiveNode>;
 
   struct null_deleter {
     void operator()(void*) {}
   };
 
+  int32_t column_index_{0};
```

Comment:
small nit, if you are reordering these you might want to move this and current two to be the last variables declared to avoid extra padding.

Reply:
Moving the int32_t to the end will not change the size of the structure on a 64-bit architecture which is now the most common architecture.